.github/workflows/scala_container_publishing.yml

```diff
@@ -1,107 +1,93 @@
-          with:
-            images: |
-              spicelabs/goatrodeo
-              ghcr.io/${{ github.repository }}
-            tags: |
-              type=semver,pattern={{version}}
-              type=semver,pattern={{major}}.{{minor}}
-              type=semver,pattern={{major}}
-              type=ref,event=branch
-              type=ref,event=pr
-              # dynamically set the branch name as a prefix
-              type=sha,prefix={{branch}}-
-              type=raw,value=latest,enable={{is_default_branch}}
-              type=ref,event=tag
-
-        - name: Stage Docker build environment from sbt
-          run: |
-            sbt Docker/stage
-        - name: Build and Push
-          id: push
-          uses: docker/build-push-action@v6
-          with:
-            context: target/docker/stage
-            push: true
-            provenance: mode=max
-            sbom: true
-            tags: ${{ steps.meta.outputs.tags }}
-            labels: ${{ steps.meta.outputs.labels }} 
-        - name: Generate artifact attestation
-          uses: actions/attest-build-provenance@v1
-          with:
-            subject-name: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
-            subject-digest: ${{ steps.push.outputs.digest }}
-            push-to-registry: true
+  push_to_registry:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'workflow_dispatch' || github.event_name == 'push' || (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') }}
+    permissions:
+      packages: write
+      contents: read
+      attestations: write
+      id-token: write
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Checkout with LFS
+        uses: actions/checkout@v4
+        with:
+          lfs: true
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
+        with:
+          java-version: '21'
+          distribution: 'adopt-hotspot'
+          cache: 'sbt'
+      - name: Set up sbt
+        uses: sbt/setup-sbt@v1
+      - name: Declare some variables
+        shell: bash
+        run: |
+          echo "short_git_hash=$(git rev-parse --short "$GITHUB_SHA")" >> "$GITHUB_ENV"
+          echo "git_branch=$(echo ${GITHUB_REF#refs/heads/})" >> "$GITHUB_ENV"
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+      - name: Log in to Github Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GH_TOKEN }}
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
```

Comment:
Does this mean not publishing to both docker and ghcr?

Reply:
Ideally we publish to both for our open source pieces.

But minimum required publishing:

* Open Source **MUST** be published to DockerHub
* Proprietary **MUST ONLY** be published to GHCR